tests/test_fips.py

```diff
@@ -189,14 +190,19 @@ def test_gnutls_binary(container_per_test: ContainerData) -> None:
 
     """
 
-    container_per_test.connection.check_output(
+    c = container_per_test.connection
+
+    c.check_output(
         "zypper --gpg-auto-import-keys -n ref && zypper -n install gcc gnutls gnutls-devel && zypper -n clean && "
         "gcc -Og -g3 fips-test-gnutls.c -Wall -Wextra -Wpedantic -lgnutls -o fips-test-gnutls && "
         "mv fips-test-gnutls /bin/fips-test-gnutls"
     )
 
-    assert "NOT" not in container_per_test.connection.check_output(
-        "gnutls-cli --fips140-mode"
+    c.check_output("gnutls-cli --fips140-mode")
+
+    assert search(
+        r"library is in FIPS140(-3|-2|) mode",
```

Comment:
do we really need to support the empty match, `|)` here? which distribution produces that output?

Reply:
Please add this to #678 instead

Reply:
@pablo-herranz ^^^

Reply:
Yes, we do as the output message might be:
- library is in FIPS140-3 mode
- library is in FIPS140-2 mode
- library is in FIPS140 mode